### DISCUSSION TITLE
Best way to write to multiple nodes at once?

### DISCUSSION THREAD

**Discussion by TheWicklowWolf** [FreeOpcUa/opcua-asyncio#1211] [Q&A] [answered]
When reading multiple values, read_values(nodes) works well. What is the equivalent for writing multiple values?
I'm using the Client from asyncua.sync

**Comment by AndreasHeine** [comment]
https://github.com/FreeOpcUa/opcua-asyncio/blob/07e6df2f2ccae4eb3649fd589496d748dc22bf0a/asyncua/sync.py#L253

**Reply by TheWicklowWolf** [reply]
When I use write_values, I get: `BadWriteNotSupported(2155020288)`

> "The server does not support writing the combination of value, status and timestamps provided."(BadWriteNotSupported)

I can write a single value using: `write_attribute(ua.AttributeIds.Value, ua.DataValue(float(value))`

What am I missing?

**Comment by TheWicklowWolf** [comment, marked as answer]
Nevermind, just used:

`write_values(nodes,[ua.DataValue(float(v)) for v in values])`

**Reply by AndreasHeine** [reply]
https://github.com/AndreasHeine/opcua-tutorial/blob/main/client/write.py

you need to use the correct varianttype

**Reply by TheWicklowWolf** [reply]
Got it working, thanks
